dizoo/trading_test/envs/trading_env.py

```diff
@@ -0,0 +1,243 @@
+from cmath import inf
+import gym
+from gym import spaces
+from gym.utils import seeding
+import numpy as np
+from enum import Enum
+import matplotlib.pyplot as plt
+from ding.envs import BaseEnv, BaseEnvTimestep
+from ding.utils import ENV_REGISTRY
+import os
+import pandas as pd
+from copy import deepcopy
+from ding.torch_utils import to_ndarray
+
+
+def load_dataset(name, index_name):
+    base_dir = os.path.dirname(os.path.abspath(__file__))
+    path = os.path.join(base_dir, 'data', name + '.csv')
+    df = pd.read_csv(path, parse_dates=True, index_col=index_name)
+    return df
+
+class Actions(Enum):
+    Double_Sell = 0
+    Sell = 1
+    Hold = 2
+    Buy = 3
+    Double_Buy = 4
+
+
+class Positions(Enum):
+    Short = -1.
+    Flat = 0.
+    Long = 1.
+
+def trans(position,  action):
+
+    if action == Actions.Sell.value:
+        
+        if position == Positions.Long:
+            return Positions.Flat, False
+            
+        if position == Positions.Flat:
+            return Positions.Short, True
+
+    if action == Actions.Buy.value:
+
+        if position == Positions.Short:
+            return Positions.Flat, False
+
+        if position == Positions.Flat:
+            return Positions.Long, True
+    
+    if action == Actions.Double_Sell.value and (position == Positions.Long or position == Positions.Flat):
+        return Positions.Short, True
+
+    if action == Actions.Double_Buy.value and (position == Positions.Short or position == Positions.Flat):
+        return Positions.Long, True
+
+    return position, False
+
+
+@ENV_REGISTRY.register('base_trading')
+class TradingEnv(BaseEnv):
+
+    def __init__(self, cfg):
+
+        self._cfg = cfg
+        #======== param to plot =========
+        self.cnt = 0 
+
+        if 'plot_freq' not in self._cfg:
+            self.plot_freq = 10
+        else:
+            self.plot_freq = self._cfg.plot_freq
+        if 'save_path' not in self._cfg :
+            self.save_path = './'
+        else:
+            self.save_path = self._cfg.save_path
+        #================================
+
+        STOCKS_GOOGL = load_dataset('STOCKS_GOOGL', 'Date')
```

Comment:
you shouldn't add data in code repo, maybe you can add a download link here (ask for @norman26625 for more help)